### PR TITLE
(WIP) NMA-410 Implement BIP270 Payment Protocol

### PR DESCRIPTION
**Pull request by tomasz-ludek**
Basic implementation of components require to handle Simplified Payment Protocol (BUP270)

WIP

### CODE REVIEW

**Comment by HashEngineering** [comment]
Travis CI fails with this message:

```
> Task :light-payment-protocol:testReleaseUnitTest
org.dash.android.lightpayprot.SimplifiedPaymentViewModelTest > when getWeather is called with valid location, then observer is updated with success FAILED
    org.mockito.exceptions.verification.WantedButNotInvoked at SimplifiedPaymentViewModelTest.kt:62
org.dash.android.lightpayprot.SimplifiedPaymentViewModelTest > when getWeather is called with invalid location, then observer is updated with failure FAILED
    org.mockito.exceptions.verification.WantedButNotInvoked at SimplifiedPaymentViewModelTest.kt:72
```

File: light-payment-protocol/src/test/java/org/dash/android/lightpayprot/SimplifiedPaymentViewModelTest.kt [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+    @Test
+    fun `when getWeather is called with invalid location, then observer is updated with failure`() = runBlocking {
+        viewModel.paymentRequest.observeForever(paymentRequestObserver)
+        viewModel.getPaymentRequest(invalidLocation)
+        delay(10)
+        verify(lightPaymentRepo).getPaymentRequest(invalidLocation)
+        verify(paymentRequestObserver, timeout(50)).onChanged(Resource.loading(null))
+        verify(paymentRequestObserver, timeout(50)).onChanged(errorResource)
+    }
```
Comment (HashEngineering) [inline light-payment-protocol/src/test/java/org/dash/android/lightpayprot/SimplifiedPaymentViewModelTest.kt]: Both tests are failing

**Comment by tomasz-ludek** [comment]
This will have to be carefully reviewed and adjusted when the right time comes

**Comment by HashEngineering** [comment]
let's keep this branch alive for review

**Pull request closed without merging**